### PR TITLE
Product.development

### PR DESCRIPTION
**Pull request by dharmarajsinh143**

### CODE REVIEW

File: app/controllers/products_controller.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      @product = Product.find(params[:id])
+    end
+
+    def product_params
+
+      params.require(:product).permit(:product_name,:product_code,:brand,:product_category,:unit,:cgst,:sgst,:product_cost,:product_price)
+    end
+  end
+
+
```
Comment (ManojParmar-MTAJ) [inline app/controllers/products_controller.rb]: fix indentation and remove extra spaces

File: app/controllers/products_controller.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  def edit
+  end
+  
+  def create
+    @product = Product.new(product_params)
+    
+    respond_to do|format|
+      if @product.save
+        format.html {redirect_to @product, notice: 'product was sussfully created.'}
+
```
Comment (RaviMariya-MTAJ) [inline app/controllers/products_controller.rb]: remove blank lines

File: app/controllers/products_controller.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  def create
+    @product = Product.new(product_params)
+    
+    respond_to do|format|
+      if @product.save
+        format.html {redirect_to @product, notice: 'product was sussfully created.'}
+        
+      else
+         format.html {render :new}
+
```
Comment (RaviMariya-MTAJ) [inline app/controllers/products_controller.rb]: same here

File: app/controllers/products_controller.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+         
+       end
+     end
+   end
+
+   def update
+    respond_to do|format|
+      if @product.update(product_params)
+        format.html {redirect_to @product, notice: 'product was sussfully updated.'}
+
```
Comment (RaviMariya-MTAJ) [inline app/controllers/products_controller.rb]: remove blank line from everywhere

File: app/views/products/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+      <ul>
+      <% product.errors.full_messages.each do |message| %>
+        <li><%= message %></li>
+      <% end %>
+      </ul>
+    </div>
+  <% end %>
+  <table border="2" align="center" cellspacing="4" cellpadding="4">
+
+<tr>
```
Comment (RaviMariya-MTAJ) [inline app/views/products/_form.html.erb]: fix indentation

File: app/views/products/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+<tr>
+  <th><strong>product_price:</strong></th>
+ <td> <%= @product.product_price%></td>
+</tr>
+
+<tr>
+<td colspan="2" style="text-align: center;"><%= link_to 'Edit', edit_product_path(@product) %></td>
+</tr>
+</table>
```
Comment (RaviMariya-MTAJ) [inline app/views/products/show.html.erb]: keep 2spaces indentation in each file

**Review by RaviMariya-MTAJ — CHANGES_REQUESTED** [review]
please review the comments and fix as suggested

File: app/views/products/_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    <li><%= message %></li>
+    <% end %>
+  </ul>
+</div>
+<% end %>
+<table class="table table stried" border="2" align="center" cellspacing="4" cellpadding="4">
+  <tr>
+    <div class="field">
+      <td><%= form.label :product_name %></td>
+      <td> <%= form.text_field :product_name, id: :product_name %></td>
```
Comment (RaviMariya-MTAJ) [inline app/views/products/_form.html.erb:16]: where are you using this id?

File: app/views/products/_form.html.erb [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    </div>
+  </tr>
+  <tr>
+    <div class="field">
+      <td><%= form.label :product_code %></td>
+      <td> <%= form.text_field :product_code, id: :product_code %></td>
```
Comment (RaviMariya-MTAJ) [inline app/views/products/_form.html.erb:22]: why you're giving id in each field?

File: app/views/products/_form.html.erb [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+<tr>
+  <div class="field">
+   <td> <%= form.label :product_cost %></td>
+   <td><%= form.number_field :product_cost, id: :product_cost%></td>
+ </div>
+</tr>
+<tr>
+  <div class="field">
+   <td> <%= form.label :product_price %></td>
+   <td><%= form.number_field :product_price, id: :product_price%></td>
```
Comment (RaviMariya-MTAJ) [inline app/views/products/_form.html.erb:64]: same everywhere

File: app/views/products/edit.html.erb [hunk 10]
```diff
@@ -0,0 +1,4 @@
+<h1 align="center">Editing Product</h1>
+<%= render 'form', product: @product %>
```
Comment (RaviMariya-MTAJ) [inline app/views/products/edit.html.erb:2]: do we need to pass `@product` here?

File: app/views/products/new.html.erb [hunk 11]
```diff
@@ -0,0 +1,3 @@
+<h1 align="center">New Product</h1>
+<%= render 'form', product: @product %>
```
Comment (RaviMariya-MTAJ) [inline app/views/products/new.html.erb:2]: same here

**Pull request merged**